src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -1362,4 +1377,109 @@ public static void makeExistingTargetSpannerDBAndTableName() {
     }
   }
 
+
+
+  @Before(order = 1, value = "@BIGTABLE_SOURCE_TEST")
+  public static void testGoogleBigtableSetup() throws IOException, URISyntaxException {
+     String projectId = PluginPropertyUtils.pluginProp("projectId");
+    instanceAdminSettings =
+            BigtableInstanceAdminSettings.newBuilder().setProjectId(projectId).build();
+    adminClient = BigtableInstanceAdminClient.create(instanceAdminSettings);
+
+    if (firstBigTableTestFlag) {
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        try {
+          BigTableClient.deleteInstance(adminClient, bigtableInstance);
+          BeforeActions.scenario.write("Bigtable instance--" + bigtableInstance + " deleted successfully");
+          bigtableInstance = StringUtils.EMPTY;
+        } catch (Exception e) {
+          if (e.getMessage().contains("NOT FOUND")) {
+            BeforeActions.scenario.write("BigTable instance--" + bigtableInstance + " does not exist.");
+          }
+        }
+      }));
+      firstBigTableTestFlag = false;
+
+      bigtableInstance = "e2e-inst-" + (int) (Math.random() * Integer.MAX_VALUE);
+      bigtableCluster = "e2e-clstr-" + (int) (Math.random() * Integer.MAX_VALUE);
+      try {
+        BigTableClient.createBigTableInstance(adminClient, bigtableInstance, bigtableCluster);
+        BeforeActions.scenario.write("BigTable instance--" + bigtableInstance + " created successfully");
+        bigTableConnection = BigTableClient.connect(projectId, bigtableInstance, null);
+        //Source table creation
+        bigtableSourceTable = "e2e-src-table-" + (int) (Math.random() * Integer.MAX_VALUE);
+        BigTableClient.createTables(bigTableConnection, bigtableSourceTable, null);
+        BeforeActions.scenario.write("sourceTable--" + bigtableSourceTable + " created successfully");
+        BigTableClient.populateData(bigTableConnection, bigtableSourceTable);
+        BeforeActions.scenario.write("sourceTable--" + bigtableSourceTable + " populated successfully");
+        PluginPropertyUtils.addPluginProp("bigtableInstance", bigtableInstance);
+        PluginPropertyUtils.addPluginProp("bigtableCluster", bigtableCluster);
+        PluginPropertyUtils.addPluginProp("bigtableSourceTable", bigtableSourceTable);
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
Please add a logger, and use LOG.error(" < SOME READABLE MESSAGE> ",e);

Reply:
For reference : see class `GCSValidationHelper`

Reply:
added